apps/backend/src/modules/users/user.repository.ts

```diff
@@ -33,9 +49,22 @@ class UserRepository implements Repository {
 	}
 
 	public async findAll(): Promise<UserEntity[]> {
-		const users = await this.userModel.query().execute();
+		const users = await this.userModel
+			.query()
+			.withGraphJoined("userDetails")
+			.execute();
+
+		return users.map((user) =>
+			UserEntity.initialize({
+				...user,
+				firstName: user.userDetails.firstName,
+				lastName: user.userDetails.lastName,
+			}),
+		);
+	}
 
-		return users.map((user) => UserEntity.initialize(user));
+	public async getByEmail(email: string): Promise<UserModel | null> {
+		return (await this.userModel.query().findOne({ email }).execute()) ?? null;
```

Comment:
```suggestion
		const user = / * ... */
		
		return user ?? null
```
remember. we can return only entity from repo